### PR TITLE
Use DebugType specified in DependencyContext to determine pdb type

### LINKED ISSUE #7006: CSharpCompiler does not honor the DebugType setting specified in the project

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #7006

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs [hunk 1]
```diff
@@ -94,11 +97,47 @@ private void EnsureOptions()
...
+            DebugInformationFormat debugInformationFormat;
+            if (string.IsNullOrEmpty(dependencyContextOptions.DebugType))
+            {
+                debugInformationFormat = SymbolsUtility.SupportsFullPdbGeneration() ?
+                    DebugInformationFormat.Pdb :
+                    DebugInformationFormat.PortablePdb;
+            }
+            else
+            {
+                // Based on https://github.com/dotnet/roslyn/blob/1d28ff9ba248b332de3c84d23194a1d7bde07e4d/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs#L624-L640
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs]: Can you please log an issue for them to expose a public parse method for this (or other guidance so we don't have to do this.

File: src/Microsoft.AspNetCore.Mvc.Razor/Resources.resx [hunk 2]
```diff
@@ -194,4 +194,7 @@
   <data name="RazorViewCompiler_ViewPathsDifferOnlyInCase" xml:space="preserve">
     <value>The following precompiled view paths differ only in case, which is not supported:</value>
   </data>
+  <data name="UnsupportedDebugInformationFormat" xml:space="preserve">
+    <value>Unsupported DebugType '{0}' specified in the application's deps file.</value>
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Resources.resx]: Let's punch this up a bit:

> The debug type specified in the dependency context could be parsed. The debug type value '{0}' is not supported.

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs [hunk 3]
```diff
@@ -54,7 +50,14 @@ public virtual CSharpCompilationOptions CSharpCompilationOptions
             }
         }
 
-        public EmitOptions EmitOptions { get; }
+        public virtual EmitOptions EmitOptions
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs]: Is this just virtual so people can override and customize if they want to?

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs]: Yup. It prevented us from suggesting overriding the property as a workaround to specifying debug type.

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs (hunk 1 above)
Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs]: https://github.com/dotnet/roslyn/issues/23997

**Pull request closed without merging**